x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/convert_to_esql_modal.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, ReactNode>>(
+    {}
+  );
+
+  const toggleDetails = useCallback(
+    (layer: ConvertibleLayer) => {
+      const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+
+      if (itemIdToExpandedRowMapValues[layer.id]) {
+        delete itemIdToExpandedRowMapValues[layer.id];
+      } else {
+        itemIdToExpandedRowMapValues[layer.id] = (
+          <EuiFlexItem>
+            <EuiCodeBlock isCopyable language="esql" paddingSize="s">
+              {layer.query}
+            </EuiCodeBlock>
+          </EuiFlexItem>
+        );
+      }
+
+      setItemIdToExpandedRowMap(itemIdToExpandedRowMapValues);
+    },
+    [itemIdToExpandedRowMap]
+  );
+
+  const columns: Array<EuiBasicTableColumn<ConvertibleLayer>> = [
+    {
+      field: 'icon',
+      name: '',
+      width: euiTheme.size.l,
+      render: (icon: string) => <EuiIcon type={icon} />,
+    },
+    {
+      field: 'name',
+      name: 'Layer',
+      width: '166px',
+      truncateText: true,
+    },
+    {
+      field: 'type',
+      name: 'Typology',
+      truncateText: true,
+      render: (type: LayerType) => typeLabels[type](1),
+    },
+    {
+      align: 'right',
+      width: euiTheme.size.xxl,
+      isExpander: true,
+      name: (
+        <EuiScreenReaderOnly>
+          <p>
+            {i18n.translate('xpack.lens.config.expandEsqlPreviewDescription', {
+              defaultMessage: 'Expand to view ES|QL query',
+            })}
+          </p>
+        </EuiScreenReaderOnly>
+      ),
+      render: (layer: ConvertibleLayer) => {
+        const isExpanded = Boolean(itemIdToExpandedRowMap[layer.id]);
+
+        return (
+          <EuiButtonIcon
+            onClick={() => toggleDetails(layer)}
+            aria-label={
+              isExpanded
+                ? i18n.translate('xpack.lens.config.collapseAriaLabel', {
+                    defaultMessage: 'Collapse',
+                  })
+                : i18n.translate('xpack.lens.config.expandAriaLabel', {
+                    defaultMessage: 'Expand',
+                  })
+            }
+            iconType={isExpanded ? 'arrowDown' : 'arrowRight'}
+            disabled={!layer.isConvertibleToEsql}
+          />
+        );
+      },
+    },
+  ];
+
+  const selection: EuiTableSelectionType<ConvertibleLayer> = {
+    selectable: (layer: ConvertibleLayer) => {
+      return layer.type === layerTypes.DATA && layer.isConvertibleToEsql;
+    },
+    selectableMessage: (selectable: boolean, layer: ConvertibleLayer) => {
+      return !selectable
+        ? layer.type === layerTypes.DATA
+          ? i18n.translate('xpack.lens.config.cannotBeSwitchedAriaLabel', {
+              defaultMessage: '{layerName} cannot be switched to query mode',
+              values: {
+                layerName: layer.name,
+              },
+            })
+          : i18n.translate('xpack.lens.config.cannotBeSwitchedAriaLabel', {
```

Comment:
This uses the same `i18n` key twice, you could fix it adding `layerName/layerTypes` in the id to differentiate.

Nit: It's not a strict rule but I remember we try to avoid nested ternaries like here. Maybe you can refactor to ifs with individual returns.